app/service/sorting.go

```diff
@@ -0,0 +1,191 @@
+	return query, NoError
+}
+
+func prepareSoringRulesAndAcceptedFields(r *http.Request, acceptedFields map[string]*FieldSortingParams,
+	defaultRules string) (string, map[string]*FieldSortingParams) {
+	newAcceptedFields := make(map[string]*FieldSortingParams, len(acceptedFields)+1)
+	for field, params := range acceptedFields {
+		newAcceptedFields[field] = params
+	}
+	if _, ok := newAcceptedFields["id"]; !ok {
+		newAcceptedFields["id"] = &FieldSortingParams{ColumnName: "ID", FieldType: "int64"}
+	}
+	var sort string
+	urlQuery := r.URL.Query()
+	if len(urlQuery["sort"]) > 0 {
+		sort = urlQuery["sort"][0]
+	} else {
+		sort = defaultRules
+	}
+	if sort == "" {
+		sort = "id"
+	}
+	return sort, newAcceptedFields
+}
+
+func parseSortingRules(sortingRules string,
+	acceptedFields map[string]*FieldSortingParams) (usedFields []string, fieldsDirections map[string]int, err error) {
+	sortStatements := strings.Split(sortingRules, ",")
+	usedFields = make([]string, 0, len(sortStatements)+1)
+	fieldsDirections = make(map[string]int, len(sortStatements)+1)
+	for _, sortStatement := range sortStatements {
+		fieldName, direction := getFieldNameAndDirectionFromSortStatement(sortStatement)
+		if fieldsDirections[fieldName] != 0 {
+			return nil, nil, fmt.Errorf("a field cannot be a sorting parameter more than once: %q", fieldName)
+		}
+		if _, ok := acceptedFields[fieldName]; !ok {
+			return nil, nil, fmt.Errorf("unknown field in sorting parameters: %q", fieldName)
+		}
+		fieldsDirections[fieldName] = direction
+		usedFields = append(usedFields, fieldName)
+	}
+	if fieldsDirections["id"] == 0 {
+		fieldsDirections["id"] = 1
+		usedFields = append(usedFields, "id")
+	}
+	return
+}
+
+func getFieldNameAndDirectionFromSortStatement(sortStatement string) (string, int) {
+	var direction int
+	if len(sortStatement) > 0 && sortStatement[0] == '-' {
+		sortStatement = sortStatement[1:]
+		direction = -1
+	} else {
+		direction = 1
+	}
+	fieldName := sortStatement
+	return fieldName, direction
+}
+
+func applyOrder(query *database.DB, usedFields []string, acceptedFields map[string]*FieldSortingParams,
+	fieldsDirections map[string]int) *database.DB {
+	usedFieldsNumber := len(usedFields)
+	orderStrings := make([]string, 0, usedFieldsNumber)
+	for _, field := range usedFields {
+		var directionString string
+		if fieldsDirections[field] == 1 {
+			directionString = "ASC"
+		} else {
+			directionString = "DESC"
+		}
+		orderStrings = append(orderStrings, acceptedFields[field].ColumnName+" "+directionString)
+	}
+	if len(orderStrings) > 0 {
+		query = query.Order(strings.Join(orderStrings, ", "))
+	}
+	return query
+}
+
+func parsePagingParameters(r *http.Request, usedFields []string,
+	acceptedFields map[string]*FieldSortingParams) ([]interface{}, error) {
+	fromValueSkipped := false
+	fromValueAccepted := false
+	fromValues := make([]interface{}, 0, len(usedFields))
+	for _, fieldName := range usedFields {
```

Comment:
I think a bit more comments all around this file would be useful to explain what you are trying to do. I think it will not be easy at all to re-enter into this code later, especially as ordering has an impact on paging.

Reply:
added more comments